wp-content/themes/core/components/comments/comment/Controller.php

```diff
@@ -0,0 +1,130 @@
+<?php
+declare( strict_types=1 );
+
+namespace Tribe\Project\Templates\Components\comments\comment;
+
+use Tribe\Libs\Utils\Markup_Utils;
+use Tribe\Project\Templates\Components\Abstract_Controller;
+
+/**
+ * Class Comment
+ */
+class Controller extends Abstract_Controller {
+
+	/**
+	 * @var int
+	 */
+	public $comment_id;
+
+	/**
+	 * @var array
+	 */
+	public $classes;
+
+	/**
+	 * @var array
+	 */
+	public $attrs;
+
+	/**
+	 * @var string
+	 */
+	public $edit_link;
+
+	/**
+	 * @var string
+	 */
+	public $reply_link;
+
+	/**
+	 * @var array
+	 */
+	public $time;
+
+	/**
+	 * Comment constructor.
+	 *
+	 * @param array $args
+	 */
+	public function __construct( array $args = [] ) {
+		$args = wp_parse_args( $args, $this->defaults() );
+		foreach ( $this->required() as $key => $value ) {
+			$args[ $key ] = array_merge( $args[ $key ], $value );
+		}
+		$this->comment_id = $args[ 'comment_id' ];
+		$this->classes    = (array) $args[ 'classes' ];
+		$this->attrs      = (array) $args[ 'attrs' ];
+		$this->edit_link  = $args[ 'edit_link' ];
+		$this->reply_link = $args[ 'reply_link' ];
+		$this->time       = (array) $args[ 'time' ];
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function defaults(): array {
+		return [
+			'comment_id' => 0,
+			'classes'    => [],
+			'attrs'      => [],
+			'edit_link'  => '',
+			'reply_link' => '',
+			'time'       => [],
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function required(): array {
+		$timestamp = get_comment_time( 'U' );
```

Comment:
This controller is inconsistent about whether it's working with the global comment or a comment that has been passed in. I think we'll be better off if it standardizes on the global comment. A lot of the logic in the comment section controller could then be moved here.

Reply:
Looked into this a bit more. While I agree, this is what I'm seeing. 
- The edit link uses a method to cut down on repeating code, if this moved to the trackback or comment controllers we'd repeat code that works here.
- There is a bit of a limitation with WP core function calls, some require a comment id and don't work inside the global comment scope without it so it's necessary. 
- The gravatar, this can be moved directly to the template but i'd guess we may want to use an image component here in the future. 

I'm not seeing much more that can be moved around. Wondering if you could provide some additional direction if you've got it?

Reply:
I've pushed an update to break apart the single comment logic from the comment section logic. Does this seem like a reasonable direction?

Reply:
This is helpful. I'll probably wait to merge this until your changes from other branches get merged in so we can do some additional cleanup on the args stuff.